crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -564,8 +559,7 @@ impl<P: ConnectionProvider> RecursorDnsHandle<P> {
         // now construct a namesever pool based off the NS and glue records
         let ns = NameServerPool::from_config(
             config_group,
-            Arc::new(self.recursor_opts()),
-            self.tls.clone(),
+            self.roots.ns.context().clone(),
```

Comment:
I'd slightly prefer that we keep an Arc of the context directly in `RecursorDnsHandle`, and clone that, just for clarity of our object model when looking at struct definitions.